src/borg/archiver.py

```diff
@@ -1243,6 +1263,13 @@ def compare_or_defer(item1, item2):
                 assert hardlink_master_seen(item2)
                 compare_items(output, item1.path, item1, item2, hardlink_masters)
 
+            if args.json_lines:
+                print_output = print_json_output
+            else:
+                print_output = print_text_output
+
+            # if we wanted sorted output (args.sort is true), then results are collected in 'output' and
+            # need to be sort them before printing. Otherwise results were already printed and 'output' is empty.
```

Comment:
if we want sorted output (args.sort is true), then results are collected in 'output' and
need to be sorted before printing. Otherwise results were already printed and 'output' is empty.